### PR TITLE
Cleanup Tabs in Blueprints page

### PR DESCRIPTION
**Pull request by KKoukiou**
…tabs

There is not any benefit to keep the active tabs in the store.
These information should be kept in the state of the component that
creates the Tabs and not somewhere in the global state, but we are using
Tabs as uncontrolled component here, so we don't need to maintain the
state of the active tabs at all.

### CODE REVIEW

**Comment by jgiardino** [comment]
Thanks for working on this! There are a few things I notice when testing the updates:

1. When I first navigate to the blueprint page, I don't see a url for the Details tab in the address bar, but when I click the Details tab, I do see a url include `/details`. I don't think this is an issue, but just wanted to mention it.
2. After I navigate to the blueprint page and click each tab, when I hit the Back button for the browser, it takes me to the previous tab I was on. If I hit the Back button again, I go forward to the last tab I was on. Hitting Back only takes me back and forth between the last two tabs.
3. If I copy and paste a url for a tab in another browser window, the contents for that page aren't loaded.
4. In the original issue, I had included this comment: 
    > This should be completed after #743, so that the urls match the updated tab labels.
    My thinking on this order is the urls we add will not change when we implement the text string changes described in #743. I'm not sure if this is a real concern or not.

**Comment by jkozol** [comment]
@jgiardino Thank you for the review. I updated the urls to reflect the text string updates.

> When I first navigate to the blueprint page, I don't see a url for the Details tab in the address bar, but when I click the Details tab, I do see a url include /details. I don't think this is an issue, but just wanted to mention it.

Thanks for pointing this out. I don't think this is an issue because it doesn't affect the ability to click on the details tab and we will be changing our default view soon.

> After I navigate to the blueprint page and click each tab, when I hit the Back button for the browser, it takes me to the previous tab I was on. If I hit the Back button again, I go forward to the last tab I was on. Hitting Back only takes me back and forth between the last two tabs.

What would your desired functionality be here? Do you think that if a user starts on the edit blueprint page then goes to the blueprint details page then changes tabs a few times they should go back to the edit blueprint page on a single Back button click?

> If I copy and paste a url for a tab in another browser window, the contents for that page aren't loaded.

I am not able to replicate this issue.

**Comment by jgiardino** [comment]
Thanks for working to finish this one up. I was able to test this out locally. I'm not seeing the issue anymore regarding ability to copy/paste a url into another browser window. And I'm fine with not seeing "/customizations" in the url when you first navigate to a blueprint page for the reasons you mention.

I am noticing some funky behavior when using some of the features built into the browser, related to the Back button, but also when I try to right click a tab and open the link associated with that tab into a new window (this issue might be due to not having an `href` value defined for the link). I captured some steps to reproduce for the first issue, that you asked about above. But also, I'm totally fine with capturing these in a separate issue to keep the scope on this PR from creeping out.

> What would your desired functionality be here? Do you think that if a user starts on the edit blueprint page then goes to the blueprint details page then changes tabs a few times they should go back to the edit blueprint page on a single Back button click?

Here is a comparison of similar actions between the PF website and Image Builder.

Steps in PF site:

1.  I open a new browser window and enter the following in the address bar: https://www.patternfly.org/v4/
2. I clicked Get started
3. I clicked Design guidelines
4. I clicked Documentation
5. I hit the Back button in the browser
6. I right-clicked the Back button, and that's the example you see below, where you can see the history of urls that existed in the address bar for this browser window, and where I'm currently at in that history.

image

Steps in Image Builder:

1.  I open a new browser window and enter the following in the address bar: https://192.168.56.102:9090/welder
2. I click on a blueprint name
3. I click the Packages tab
4. I click the Images tab
5. I click the Edit Packages button
6. I click the Back button in the browser
7. I right-clicked the Back button, and that's the example you see below.

In this case, what you can't see is that the Forward button is disabled and the Edit Packages page isn't represented in this history. But also, there are two instances for each URL listed, and hitting the Back button doesn't traverse this history as expected. I would suggest trying this yourself and hovering over each item in the browser history for the Back button to see the actual urls being listed. As I mentioned above, I'm totally fine capturing this as a separate issue, but wanted to mention it here in case it's obvious what the fix is and it's easy to include.  

image

File: pages/blueprint/index.js [hunk 1]
```diff
@@ -311,7 +311,7 @@ class BlueprintPage extends React.Component {
       users = blueprint.customizations.user;
     }
     const pathSuffix = cockpit.location.path[cockpit.location.path.length - 1];
-    const activeKey = ["details", "selected-components", "images"].includes(pathSuffix) ? pathSuffix : undefined;
+    const activeKey = ["customizations", "packages", "images"].includes(pathSuffix) ? pathSuffix : undefined;
```
Comment (KKoukiou) [inline pages/blueprint/index.js]: This change should be done when introducing this line in the previous commit and not here https://github.com/weldr/cockpit-composer/pull/752/commits/e9359dbb62a82d8571f55195569b9b99edb0d84a

File: pages/blueprint/index.js [hunk 2]
```diff
@@ -381,7 +381,7 @@ class BlueprintPage extends React.Component {
           onSelect={eventId => cockpit.location.go(["blueprint", blueprint.name, eventId])}
           id="blueprint-tabs"
         >
-          <Tab eventKey="details" title="Details">
+          <Tab eventKey="customizations" title={formatMessage(messages.customizationsTitle)}>
```
Comment (KKoukiou) [inline pages/blueprint/index.js]: Same for this.

**Pull request merged**